app/models/gemini_connection.rb

```diff
@@ -64,27 +64,30 @@ def fetch_refresh_token
   end
 
   def sync_connection!
-    return if access_token.blank?
+    result = refresh_authorization!
 
-    # If our access token has expired then we should refresh.
-    if access_token_expired?
-      refresh_authorization!
-    end
+    case result
+    when GeminiConnection
+      users = Gemini::Account.find(token: access_token).users
+      user = users.find { |u| u.is_verified && u.status == "Active" }
 
-    users = Gemini::Account.find(token: access_token).users
-    user = users.find { |u| u.is_verified && u.status == "Active" }
+      # If we couldn't find a verified account we'll take the first user.
+      user ||= users.first
 
-    # If we couldn't find a verified account we'll take the first user.
-    user ||= users.first
+      update(
```

Comment:
@tsmartt DO we really want to run this every time the user logs in?  I understand the refresh_authorization! piece but syncing over their data on every login?

Reply:
Once a month should be fine, in our job